email_composer/src/utils/EmailUtils.js

```diff
@@ -347,7 +348,20 @@ export const parseEmailAddress = emailAddress => {
   email = email.toLowerCase();
   const emailTag = isEmailTag ? email : `<${email}>`;
   const complete = `${name || ''} ${emailTag}`;
-  return { name, email, complete: complete.trim() };
+  const domain = email.split('@')[1];
+  const form = emailRegex.test(email)
+    ? email.includes(`@${appDomain}`)
```

Comment:
- If you are an enterprise user, better check if the email contains your own domain to avoid unnecessary calls to the server. Should be common for users in the same business to write emails between themself so this should leverage network traffic.

- Also having a cache for enterprise domains would be useful